test-avocado/checklogin.py

```diff
@@ -0,0 +1,146 @@
+        b.wait_text('#content-user-name', 'Administrator')
+
+        b.click('#go-account')
+        b.enter_page("account")
+        b.wait_text ("#account-user-name", "admin")
+
+    def curl_auth(self, url, userpass):
+        header = "Authorization: Basic " + base64.b64encode(userpass)
+        return process.run("/usr/bin/curl -s -k -D - --header %s http://%s:9090%s " %  (header, "localhost", url), ignore_status=True)
+
+    def curl_auth_code(self, url, userpass):
+        lines = self.curl_auth(url, userpass).stdout.splitlines()
+        assert len(lines) > 0
+        tokens = lines[0].split(' ', 2)
+        assert len(tokens) == 3
+        self.log.debug(tokens)
+        return int(tokens[1])
+
+    def Raw(self):
+        time.sleep(0.5)
+        self.assertEqual(self.curl_auth_code ('/login', ''), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'foo:'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'foo:bar\n'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'foo:bar:baz'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', ':\n\n'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'admin:bar'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'foo:bar'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'admin:' + 'x' * 4000), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'x' * 4000 + ':bar'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'a' * 4000 + ':'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'a' * 4000 + ':b\nc'), 401)
+        self.assertEqual(self.curl_auth_code ('/login', 'a' * 4000 + ':b\nc\n'), 401)
+        j=Journalctl()
+        j.allow_journal_messages ("Returning error-response ... with reason .*", "pam_unix\(cockpit:auth\): authentication failure; .*", "pam_unix\(cockpit:auth\): check pass; user unknown", "pam_succeed_if\(cockpit:auth\): requirement .* not met by user .*", "couldn't parse login input: Malformed input", "couldn't parse login input: Authentication failed")
+        j.allow_restart_journal_messages
+        j.check_journal_messages()
+
+    def setup(self):
+# Setup users and passwords
+        process.run("useradd %s -c 'Barney Bär'" % username, shell=True, ignore_status=True)
+        process.run("echo abcdefg | passwd --stdin %s" % username, shell=True)
+
+        process.run("useradd %s -c 'Administrator'" % "admin", shell=True, ignore_status=True)
+        process.run("gpasswd wheel -a %s" % "admin", shell=True, ignore_status=True)
+        process.run("/bin/cp /etc/pam.d/cockpit{,.old}", shell=True, ignore_status=True)
+        process.run("echo foobar | passwd --stdin %s" % "admin", shell=True)
+
+
+        # Setup a special PAM config that disallows non-wheel users
+        process.run("echo '%s' > /etc/pam.d/cockpit" % admins_only_pam, shell=True)
+
+        process.run("systemctl start cockpit", shell=True ,ignore_status=True)
```

Comment:
another thing why I'm sometimes ignoring statuses is because I don't know previous state of system, like here.
In case service was already started it will fail, and I'm not sure with handling these situation in code via some IF or somehow when this ignoring can does almost same thing in simpler way finally. But I undestand I can improve that somehow.